zero-index get/setindex(::ReinterpretArray) require a length of 1 (#58814)
JuliaLang/julia: NEWS.md

  - The `hash` algorithm and its values have changed. Most `hash` specializations will remain correct and require no action. Types that reimplement the core hashing logic independently, such as some third-party string packages do, may require a migration to the new algorithm. ([#57509])

> * indexless `getindex` and `setindex!` (i.e. `A[]`) on `ReinterpretArray` of nonzero dimensionality now correctly throw a `BoundsError`. ([#58814])

Reviewer: ```suggestion
* Indexless `getindex` and `setindex!` (i.e. `A[]`) on `ReinterpretArray` now correctly throw a `BoundsError` when there is more than one element. ([#58814])
```
